src/python/pants/base/build_environment.py

```diff
@@ -23,12 +23,12 @@ def pants_release():
           .format(version=pants_version()))
 
 
-def get_buildroot():
-  """Returns the pants build root, calculating it if needed.
+def get_buildroot(*, sentinel_filename: str = "pants") -> str:
+  """Returns the Pants build root, calculating it if needed.
```

Comment:
It would be good to explain that this is only settable for unit testing, or to make it more explicitly settable.

If we have essentially changed things to make the sentinel value to look for mutable, maybe we should have an explicit `set_sentinel` method that resets the buildroot and forces it to be recomputed?

Reply:
> It would be good to explain that this is only settable for unit testing, or to make it more explicitly settable.

Went with explaining. IMO better to not make something like this public unless someone requests it, because it increases ever so slightly the complexity of program.

> If we have essentially changed things to make the sentinel value to look for mutable, maybe we should have an explicit set_sentinel method that resets the buildroot and forces it to be recomputed?

Sounds good. I made this change with a slight modification that it only resets the buildroot but does not recompute it. Less side-effects that way. I agree that the `path` becomes invalid, but it shouldn't be recomputed unless a user explicitly requests it.